impeller/geometry/rect.h

```diff
@@ -130,7 +130,8 @@ struct TRect {
   }
 
   [[nodiscard]] constexpr bool Contains(const TRect& o) const {
```

Comment:
We want `a.Contains(a) == true`, right?

Reply:
Yes indeed, we assume this inclusivity in our current usages

Reply:
I only ask because the Contains(Point) uses < operators for the right and bottom axises.

Reply:
Another issue with these methods is that empty can contain another rectangle and vice versa. I'm planning to address all of these in a rewrite now that the fields are hidden, but I'm just starting on that now and if this PR fixes an existing practical issue, then we should proceed with it.

Reply:
The left and top edges are considered "inside", the right and bottom edges are considered "outside".

For points you want `>= left/top` and `< right/bottom` to match that rule.

For rects, both have the same insideness and so matching edges always have the same containment as each other. In the case of right and bottom, those edges are the non-inclusive limit of what each rectangle considers "inside" and so if they are equal then all of the points inside one of them are also inside the other (wrt that one edge).